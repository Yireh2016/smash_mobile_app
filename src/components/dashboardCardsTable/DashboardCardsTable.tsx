/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Card } from '../../types/Card';

import CardListItem from '../cardListItem/CardListItem';

import TableRow from '../tableRow/TableRow';

import { P1, P2 } from '../texts/Texts';
import formatMoneyValue from '../../helpers/formatMoneyValue/formatMoneyValue';
import formatPercentage from '../../helpers/formatPercentage/formatPercentage';
import CardNameMaskPair from '../cardNameMaskPair/CardNameMaskPair';
import mappedCards from '../../helpers/mappedCards/mappedCards';

interface DashboardCardsTableProps {
  cards: Card[];
}

const styles = StyleSheet.create({
  cardsLayout: {
    marginTop: 16,
  },
  firstCardsLayout: {
    marginTop: 10,
  },
});
const DashboardCardsTable: React.FunctionComponent<DashboardCardsTableProps> = ({
  cards,
}) => {
  const cardsMap = cards.map((card: Card, i: number) => {
    return (
      <View
        key={i}
        style={i === 0 ? styles.firstCardsLayout : styles.cardsLayout}
      >
        <CardListItem>
          <View
            style={{
              paddingHorizontal: 13,
              paddingVertical: 18,
            }}
          >
            <TableRow
              columns={[
                {
                  component: (
                    <CardNameMaskPair
                      maxLentgh={15}
                      card={mappedCards([card])[0]}
                    />
                  ),
                  align: 'left',
                  width: 60,
                },
                {
                  component: (
                    <P1 isBold={true}>{`${formatMoneyValue(
                      card.current_balance
                    )}`}</P1>
                  ),
                  align: 'left',
                  width: 20,
                },
                {
                  component: (
                    <P1 isBold={true}>{`${formatPercentage(
                      card.credit_utilization_percentage
                    )}`}</P1>
                  ),
                  align: 'right',
                  width: 20,
                },
              ]}
            />
          </View>
        </CardListItem>
      </View>
    );
  });
  return (
    <>
      <View style={{ paddingRight: 16 }}>
        <TableRow
          columns={[
            { component: <P2>Card</P2>, align: 'left', width: 60 },
            { component: <P2>Balance</P2>, align: 'left', width: 20 },
            { component: <P2>Utilization</P2>, align: 'right', width: 20 },
          ]}
        />
      </View>
      {cardsMap}
    </>
  );
};

export default DashboardCardsTable;
