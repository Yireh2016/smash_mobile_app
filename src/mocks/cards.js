export default [
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '2267',
    current_balance: 500,
    credit_utilization_percentage: 10,
    statement_balance: 2650,
    purchase_apr_percentage: 19,
    min_payment: 65,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 19,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 3100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9376',
    current_balance: 2000,
    credit_utilization_percentage: 25,
    statement_balance: 1030,
    purchase_apr_percentage: 21,
    min_payment: 37,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 21,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9085',
    current_balance: 3000,
    credit_utilization_percentage: 10,
    statement_balance: 1820,
    purchase_apr_percentage: 22,
    min_payment: 42,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 22,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1900,
      },
    ],
  },
];
