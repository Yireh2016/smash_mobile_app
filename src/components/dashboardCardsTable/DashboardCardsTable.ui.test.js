import * as React from 'react';
import renderer from 'react-test-renderer';
import DashboardCardsTable from './DashboardCardsTable';
import cards from '../../mocks/cards';
it('DashboardCardsTable renders correctly', () => {
  const ComponentProps = {
    cards,
  };
  const Element = () => {
    return <DashboardCardsTable {...ComponentProps} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
