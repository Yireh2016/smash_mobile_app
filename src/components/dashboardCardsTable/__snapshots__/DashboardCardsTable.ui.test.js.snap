// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCardsTable renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "paddingRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "width": "60%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            }
          }
        >
          Card
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            }
          }
        >
          Balance
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            }
          }
        >
          Utilization
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#142836",
          "borderRadius": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 13,
            "paddingVertical": 18,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "60%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 16,
                  }
                }
              >
                Citi
              </Text>
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 16,
                  }
                }
              >
                 (2267)
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              $500
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              10%
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#142836",
          "borderRadius": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 13,
            "paddingVertical": 18,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "60%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 16,
                  }
                }
              >
                Citi
              </Text>
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 16,
                  }
                }
              >
                 (9376)
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              $2,000
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              25%
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#142836",
          "borderRadius": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 13,
            "paddingVertical": 18,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "60%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Bold",
                    "fontSize": 16,
                  }
                }
              >
                Citi
              </Text>
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 16,
                  }
                }
              >
                 (9085)
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              $3,000
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "width": "20%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 16,
                }
              }
            >
              10%
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
